python/tvm/relay/backend/contrib/tidl.py

```diff
@@ -0,0 +1,1407 @@
+            if len(output_fp) != len(out_data_q):
+                return False
+            output_signed = []
+            output_scale  = []
+            for i in range(len(output_fp)):
+                output_signed.append(int(np.amin(output_fp[i]) < 0))
+            for i in range(len(out_data_q)):
+                output_scale.append(round(out_data_q[i]/255.0,5))  # 255 is TIDL implementation specific
+            
+            # Generate subgraph configuration file
+            subgraph_cfg_gen(self.artifacts_folder, subgraph_id, self.data_layout, 
+                            input_scale, input_signed, output_scale, output_signed)
+        return True
+
+
+
+class TIDLCompiler:
+    """TIDL compiler module.
+
+    This module tries to compile a given Relay IR graph to deploy on devices with TIDL.
+    If compilation for TIDL succeeds, artifacts for heterogeneous compute with TIDL
+    will be generated.
+
+    Parameters
+    ----------
+    platform : string
+        The platform to deploy the graph on.
+    version : tuple
+        The Processor-SDK version for the platform.
+    **kwargs : keyword arguments to pass what's needed for Relay IR graph conversion
+        num_tidl_subgraphs : int
+            Number of subgraphs to run on TIDL
+        data_layout : string
+            Data layout, "NCHW" or "NHWC"
+        tidl_tools_path : string
+            Folder to TIDL tools
+        artifacts_folder : string
+            Folder to hold TIDL artifacts
+    """
+
+    def __init__(self, platform, version, **kwargs):
+        if platform == "AM57" and version >= (6,3):
+            for key in ('num_tidl_subgraphs', 'data_layout', 'artifacts_folder', 'tidl_tools_path'):
+                if key in kwargs:
+                    setattr(self, key, kwargs[key])
+            self.tidl_target = "tidl"
+        else:
+            sys.exit("Unsupported TIDL platform or version!")
+
+    def enable(self, mod_orig, params, input):
+        """ Enable TIDL compilation
+
+        This function tries to partition and compile the given Relay IR graph.
+        If it succeeds, artifacts for heterogeneous compute with TIDL will be 
+        generated, and the partitioned graph will be returned. Otherwise, it will 
+        return None.
+
+        Parameters
+        ----------
+        mod_orig : tvm.relay.Module 
+            Original Relay IR graph
+        params : dict of str to tvm.NDArray
+            The parameter dict to be used by relay
+        input: dictionary
+            A dictionary where the key in input name and the value is input tensor
+
+        Returns
+        -------
+        mod : tvm.relay.Module 
+            Paritioned graph with subgraphs to run with TIDL
+        status: int
+            Status of TIDL compilation: 
+                1  - compilation success
+                -1 - compilation failure
+                0  - no compilation due to missing TIDL tools
+        """
+
+        if len(input) > 1:

```

Comment:
Use `==`.

Reply:
@comaniac, can you please elaborate more why "==" should be used?

Reply:
This check will be removed.